rootfs/etc/nginx/lua/test/balancer/ewma_test.lua

```diff
@@ -52,40 +70,61 @@ describe("Balancer ewma", function()
 
   describe("sync()", function()
     local backend, instance
-
-    before_each(function()
-      backend = {
-        name = "my-dummy-backend", ["load-balance"] = "ewma",
-        endpoints = { { address = "10.184.7.40", port = "8080", maxFails = 0, failTimeout = 0 } }
+    local store_ewma_stats = function(endpoint_string, ewma, touched_at)
+      ngx.shared.balancer_ewma:set(endpoint_string, ewma)
+      ngx.shared.balancer_ewma_last_touched_at:set(endpoint_string, touched_at)
+    end
+    local assert_ewma_stats = function(endpoint_string, ewma, touched_at)
+      assert.are.equals(ewma, ngx.shared.balancer_ewma:get(endpoint_string))
+      assert.are.equals(touched_at, ngx.shared.balancer_ewma_last_touched_at:get(endpoint_string))
+    end
+
+    backend = {
+      name = "namespace-service-port", ["load-balance"] = "ewma",
+      endpoints = {
+        { address = "10.10.10.1", port = "8080", maxFails = 0, failTimeout = 0 },
+        { address = "10.10.10.2", port = "8080", maxFails = 0, failTimeout = 0 },
+        { address = "10.10.10.3", port = "8080", maxFails = 0, failTimeout = 0 },
       }
-      instance = balancer_ewma:new(backend)
-    end)
+    }
+    store_ewma_stats("10.10.10.1:8080", 0.2, 1543238266)
+    store_ewma_stats("10.10.10.2:8080", 0.3, 1543238269)
+    store_ewma_stats("10.10.10.3:8080", 1.2, 1543238226)
 
-    it("does nothing when endpoints do not change", function()
-      local new_backend = {
-        endpoints = { { address = "10.184.7.40", port = "8080", maxFails = 0, failTimeout = 0 } }
-      }
-
-      instance:sync(new_backend)
-    end)
+    instance = balancer_ewma:new(backend)
 
-    it("updates endpoints", function()
-      local new_backend = {
-        endpoints = {
-          { address = "10.184.7.40", port = "8080", maxFails = 0, failTimeout = 0 },
-          { address = "10.184.97.100", port = "8080", maxFails = 0, failTimeout = 0 },
-        }
-      }
+    it("does not reset stats when endpoints do not change", function()
+      local new_backend = util.deepcopy(backend)
 
       instance:sync(new_backend)
+
       assert.are.same(new_backend.endpoints, instance.peers)
+
+      assert_ewma_stats("10.10.10.1:8080", 0.2, 1543238266)
+      assert_ewma_stats("10.10.10.2:8080", 0.3, 1543238269)
+      assert_ewma_stats("10.10.10.3:8080", 1.2, 1543238226)
     end)
 
-    it("resets stats", function()
+    it("updates peers, deletes stats for old endpoints and sets average ewma score to new ones", function()
       local new_backend = util.deepcopy(backend)
-      new_backend.endpoints[1].maxFails = 3
+
+      -- existing endpoint 10.10.10.2 got deleted
```

Comment:
I think your comment here doesn't match the actual value underneath

Reply:
this specific line or the line below?

I agree that the line below does not match

Reply:
```
-- existing endpoint 10.10.10.2 got deleted
-- and replaced with 11.10.10.2
```

and 
```
new_backend.endpoints[2].address = "10.10.10.4"
```